setup.sh

```diff
@@ -16,8 +16,15 @@ if [ "0$ID" -ne 0 ]
   exit
 fi
 
-# Ensure the script is running on something it can work with
-if [ ! -f /etc/os-release ]; then
+# Ensure the 'docker' and 'docker-compose' commands are available
+# and if not, ensure the script can install them
+SKIP_DOCKER_INSTALL=no
+if [ -x "$(command -v docker)" ]; then
+  # The first condition is 'docker-compose (v1)' and the second is 'docker compose (v2)'.
+  if [ -x "$(command -v docker-compose)" ] || (docker compose 1> /dev/null 2>& 1 && [ $? -eq 0 ]); then
```

Comment:
You could probably improve this check a bit by combining the two checks into one.

Something like this:

```
if [ -x "..test1.." -a -x "..test2.." ]; then ...
```

The concept above is the `-a` (AND) can be use to join tests:  "if A ***AND*** B are both true".

There's also the `-o` (OR) operation, for testing if one of a series is true, but that's probably not useful for this PR.

Reply:
That was my first thought, but wouldn't conditional statements be too complicated?
And the second if statement is using an OR condition to check for both docker-compose v1 and v2.

Reply:
Good point.  Yeah, complication just leads to mistakes.  Let's go with your approach, as it should mean future people can maintain it pretty easily too. :smile:

Reply:
That's great, can we merge now? :)